File: python/docs_code_snippets/spans-update.py
from datetime import datetime, timezone
from opperai import Opper

opper = Opper()

# Setup
span = opper.spans.create(
    name="my-pipeline",
    start_time=datetime.now(timezone.utc).isoformat(),
)

# --- docs ---
opper.spans.update(
    span.id,
    output="Pipeline completed successfully",
    end_time=datetime.now(timezone.utc).isoformat(),
    meta={"total_calls": 3},
)
# --- /docs ---

# Cleanup
opper.spans.delete(span.id)

File: typescript/docs_code_snippets/spans-update.ts
import { Opper } from "opperai";

const opper = new Opper();

// setup
const created = await opper.spans.create({ name: "docs-snippet-span" });
const spanId = created.id;
// /setup

// --- docs ---
await opper.spans.update(spanId, {
  output: "Pipeline completed successfully",
  end_time: new Date().toISOString(),
  meta: { total_calls: 3 },
});
// --- /docs ---

console.log("Span updated");
// cleanup
await opper.spans.deleteSpan(spanId);
